CMakeLists.txt

```diff
@@ -37,11 +41,35 @@ set(CMAKE_Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/swift)
 set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreadedDLL)
 set(CMAKE_POSITION_INDEPENDENT_CODE ${BUILD_SHARED_LIBS})
 
+# Toolchain vended dependencies
 find_package(dispatch CONFIG)
 find_package(Foundation CONFIG)
 
-find_package(SwiftASN1 CONFIG REQUIRED)
-find_package(SwiftCrypto CONFIG REQUIRED)
+include(FetchContent)
+
+set(_CERT_SAVED_BUILD_EXAMPLES ${BUILD_EXAMPLES})
+set(_CERT_SAVED_BUILD_TESTING ${BUILD_TESTING})
+
+find_package(SwiftASN1 CONFIG)
+if(NOT SwiftASN1_FOUND)
+  message("-- Vending swift-asn1")
+  FetchContent_Declare(ASN1
+    GIT_REPOSITORY https://github.com/apple/swift-asn1
+    GIT_TAG 1.1.0)
```

Comment:
It is highly likely that we will forget to keep this in-sync with `Package.swift`. Would it be better to require users of this feature to set a version environment variable so they don't use a wrong version by accident?

Reply:
That does make it a lot more complicated to use.  The intention here is to enable the use of the CMake based build to debug LSP, which requires SPM, which requires this package.

Reply:
@yim-lee Can we amend the soundness script to validate that these two match?

Reply:
If we look at the rule for `swift-crypto`:

```
.package(url: "https://github.com/apple/swift-crypto.git", "2.5.0" ..< "4.0.0"),
```

What's the `swift-crypto` version that we would extract from this for comparison/validation?

If we extract from `Package.resolved` instead, the resolved version may not necessarily match the one in this file. This might be annoying but could work I suppose? I don't think we do `swift build` or `swift package resolve` in the soundness check though.

Reply:
Yeah, these are all good questions. I'm just wondering if we can get to a local maximum of "least likely to be outside the band of reasonableness".